code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -180,21 +203,39 @@ public Object call() throws Exception {
         }
     }
 
-    private Instance createInstance(Service service, int i) {
-        Environment env = objectManager.findOne(Environment.class, ENVIRONMENT.ID, service.getEnvironmentId());
-        String instanceName = String.format("%s_%s_%d", env.getName(), service.getName(), i + 1);
+
+    private List<Long> getServiceNetworks(Service service) {
+        List<Long> ntwkIds = new ArrayList<>();
+        long networkId = sdServer.getServiceNetworkId(service);
+        ntwkIds.add(networkId);
+        return ntwkIds;
+    }
+
+    protected Long getImage(String imageUuid) {
+        Image image;
+        try {
+            image = storageService.registerRemoteImage(imageUuid);
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to get image [" + imageUuid + "]");
+        }
+
+        return image == null ? null : image.getId();
+    }
+
+
+    protected Instance createInstance(Service service, int i, Map<String, Object> launchConfigData, Long imageId,
+            List<Long> networkIds) {
+        String instanceName = sdServer.getInstanceName(service, i);
         Instance instance = objectManager.findOne(Instance.class, INSTANCE.NAME, instanceName,
```

Comment:
Does `ServiceExposeMapDao.getServiceInstance` do the same thing (or close enough to the same thing that it could/should be used here?